src/crimson/os/seastore/onode_manager/staged-fltree/tree.h

```diff
@@ -265,10 +265,10 @@ class Btree {
     value_config_t vconf{value_builder.get_header_magic(), _vconf.payload_size};
     return seastar::do_with(
       full_key_t<KeyT::HOBJ>(obj),
-      [this, &t, vconf](auto& key) -> eagain_ifuture<std::pair<Cursor, bool>> {
+      [this, &t, vconf, &obj](auto& key) -> eagain_ifuture<std::pair<Cursor, bool>> {
         ceph_assert(key.is_valid());
-        return get_root(t).si_then([this, &t, &key, vconf](auto root) {
-          return root->insert(get_context(t), key, vconf, std::move(root));
+        return get_root(t).si_then([this, &t, &key, vconf, &obj](auto root) {
+          return root->insert(get_context(t, get_hint(obj)), key, vconf, std::move(root));
```

Comment:
I'm fine to imbed the hint in the `context_t` for now which is calculated by the inserting/removing object.

But a more appropriate hint would be from the first key of the allocating node during split I think, which finally won't be needed in the `context_t`

Reply:
Yeah, that's my thought as well.  I think we have access to the full current pivot key whenever we're allocating a staged-fltree node, so we shouldn't need to pass a hint through like this.

Reply:
Perhaps we should merge the other portions first and tackle staged-fltree in a follow up PR?

Reply:
Yeah, agree.

Reply:
The current version still isn't deriving its hint from the node being allocated.  Can we just drop that commit for now and revisit the staged-fltree changes later?

Reply:
let's save hint in context_t now, otherwise, we should transfer hint as parameter in callstack before calling allocate since we already transfer context_t as parameter in callstack.

Reply:
I see, I will check the code.

Reply:
@athanatos @cyx1231st I remove hint from context_t and get hint from key before calling allocate in the last commit. please review